afew/tests/test_mailmover.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: ISC
+# Copyright (c) Johannes Löthberg <johannes@kyriasis.com>
+
+import datetime
+import email
+from email.utils import format_datetime, localtime, make_msgid
+import mailbox
+import notmuch
+import os
+import shutil
+import tempfile
+import unittest
+
+
+def get_notmuch_db(db_dir):
+    return notmuch.Database(db_dir, mode=notmuch.Database.MODE.READ_WRITE)
+
+def create_mail(msg, maildir, notmuch_db, tags, old=False):
+    if old:
+        date = format_datetime(localtime() - datetime.timedelta(days=20))
+    else:
+        date = format_datetime(localtime())
+    msgid = make_msgid()
+    email_message = email.message_from_string("""\
+Date: {date}
+From: you <you@example.org
+To: me <me@example.com>
+Message-ID: {msgid}
+
+{msg}""".format(date=date, msgid=msgid, msg=msg))
+    maildir_message = mailbox.MaildirMessage(email_message)
+    message_key = maildir.add(maildir_message)
+
+    fname = os.path.join(maildir._path, maildir._lookup(message_key))
+    # notmuch 0.26 added index_file and deprecated add_message.
+    if hasattr(notmuch.Database, 'index_file'):
+        (notmuch_msg, status) = notmuch_db.index_file(fname)
+    else:
+        (notmuch_msg, status) = notmuch_db.add_message(fname)
+    for tag in tags:
+        notmuch_msg.add_tag(tag, False)
+
+    return (msgid[1:-1], msg)
```

Comment:
what are we doing here, and why?

Reply:
Stripping the angle brackets off of the message ID that `make_msgid` generated, since the notmuch `get_message_id` function doesn't include them.  I switched to using `strip('<>')` for this instead though.